01/app.js

```diff
@@ -7,7 +7,7 @@ function getProductList() {
 
 function getTotalPrice(productList) {
     return productList.reduce( (product, acc) => {
-        return acc + (product.price * product.count);
+        return (((acc.price - (acc.price * acc.discount)) * acc.count) + ((product.price - (product.price * product.discount)) * product.count)).toFixed(2);
```

Comment:
Nie dałoby się prościej? Dodatkowo, czy to rozwiązanie zadziała jak tablica będzie pusta?

Reply:
Dało się prościej :) Teraz z pustą tablicą działa, choć wiem, że to nie idealny sposób. Nie mogę dojść do tego, jak miałoby to wyglądać z podstawieniem 0 za acumulator, tak jak zwykle w zapisie arr.reduce(callback(), 0). Wtedy przy nie-pustej tablicy otrzymuję NaN. Nie potrafię znaleźć dlaczego :x